tests/models/audio_spectogram_transformer/test_modeling_audio_spectogram_transformer.py

```diff
@@ -0,0 +1,263 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Testing suite for the PyTorch AudioSpectogramTransformer model. """
+
+import inspect
+import unittest
+
+from huggingface_hub import hf_hub_download
+from transformers import AudioSpectogramTransformerConfig
+from transformers.testing_utils import require_torch, require_torchaudio, slow, torch_device
+from transformers.utils import cached_property, is_torch_available, is_torchaudio_available
+
+from ...test_configuration_common import ConfigTester
+from ...test_modeling_common import ModelTesterMixin, floats_tensor, ids_tensor
+
+
+if is_torch_available():
+    import torch
+    from torch import nn
+
+    from transformers import AudioSpectogramTransformerForSequenceClassification, AudioSpectogramTransformerModel
+    from transformers.models.audio_spectogram_transformer.modeling_audio_spectogram_transformer import (
+        AUDIO_SPECTOGRAM_TRANSFORMER_PRETRAINED_MODEL_ARCHIVE_LIST,
+    )
+
+
+if is_torchaudio_available():
+    import torchaudio
+
+    from transformers import AudioSpectogramTransformerFeatureExtractor
+
+
+class AudioSpectogramTransformerModelTester:
+    def __init__(
+        self,
+        parent,
+        batch_size=13,
+        patch_size=2,
+        time_dimension=24,
+        frequency_dimension=16,
+        is_training=True,
+        use_labels=True,
+        hidden_size=32,
+        num_hidden_layers=5,
+        num_attention_heads=4,
+        intermediate_size=37,
+        hidden_act="gelu",
+        hidden_dropout_prob=0.1,
+        attention_probs_dropout_prob=0.1,
+        type_sequence_label_size=10,
+        initializer_range=0.02,
+        scope=None,
+        frequency_stride=2,
+        time_stride=2,
+    ):
+        self.parent = parent
+        self.batch_size = batch_size
+        self.patch_size = patch_size
+        self.time_dimension = time_dimension
+        self.frequency_dimension = frequency_dimension
+        self.is_training = is_training
+        self.use_labels = use_labels
+        self.hidden_size = hidden_size
+        self.num_hidden_layers = num_hidden_layers
+        self.num_attention_heads = num_attention_heads
+        self.intermediate_size = intermediate_size
+        self.hidden_act = hidden_act
+        self.hidden_dropout_prob = hidden_dropout_prob
+        self.attention_probs_dropout_prob = attention_probs_dropout_prob
+        self.type_sequence_label_size = type_sequence_label_size
+        self.initializer_range = initializer_range
+        self.scope = scope
+        self.frequency_stride = frequency_stride
+        self.time_stride = time_stride
+
+        # in AudioSpectogramTransformer, the seq length equals the number of patches + 2 (we add 2 for the [CLS] and distillation tokens)
+        test_input = torch.randn(1, 1, self.frequency_dimension, self.time_dimension)
+        test_projection = nn.Conv2d(
```

Comment:
Again, don't think it's strictly necessary to have to perform this dummy conv operation!